HeuristicNew.java

```diff
@@ -2,56 +2,34 @@
 	public static final int NUM_FEATURES = 6;
 	public static final boolean db = false; // set true to turn debug statements on
     public static final boolean db2 = false; // set true to debug rows eliminated (recommended set db = false)
-	static double[] weights;
-	static int weightCounter;
+	double[] weights;
+	int weightCounter;
+    TempState s;
+
+    public HeuristicNew(TempState s, double[] inputWeights) {
```

Comment:
Very standard change of static methods to public in order to enable OOP. Looks good.